### PR TITLE
Add support for specifying the WebSequenceDiagrams.com API key

### PR DESCRIPTION
**Pull request by icedawn**
Added support for specifying the WebSequenceDiagrams.com API key on the command line or environment variable (WSD_APIKEY).

### CODE REVIEW

File: lib/wsd.js [hunk 1]
```diff
@@ -19,7 +19,7 @@ const styles = [
   'patent']
 
 class WSD {
-  static async diagramURL (message, style = 'default', format = 'png') {
+  static async diagramURL (message, style = 'default', format = 'png', apikey = '') {
```
[the remark this answers is not in the document]
Comment (icedawn) [reply lib/wsd.js]: Cool, got it.

File: bin/wsd_get [hunk 2]
```diff
@@ -24,6 +24,10 @@ const argv = opt
   .string('s')
   .alias('s', 'style')
   .describe('s', 'Output style (one of: [' + wsd.styles.join(', ') + '])')
+  .string('k')
+  .alias('k', 'key')
+  .describe('k', 'WebSequenceDiagrams API key.  Key can also be specified with WSD_APIKEY environment variable.  Requires premium account, see https://www.websequencediagrams.com/order.html')
+  .default('k', process.env.WSD_APIKEY)
```
[the remark this answers is not in the document]
Comment (icedawn) [reply bin/wsd_get]: Excellent.  Looks like yargs will handle this well.  The WSD prefix seems appropriate.   We'd have to choose whether the arg is "key" or "apikey" which would map to the environment variables WSD_KEY or WSD_APIKEY respectively.  I'd lean towards "apikey" to be more descriptive.

**Comment by icedawn** [comment]
I'll look into writing an appropriate unit test to check the API key path to get the coverage back up.

File: bin/wsd_get (hunk 2 above)
[the remark this answers is not in the document]
Comment (hildjj) [reply bin/wsd_get]: `WSD_APIKEY` feels just right.

**Comment by icedawn** [comment]
Still looking at adding unit tests for the new functionality and to get the code coverage back up.

**Comment by mikelax** [comment]
I'm just came across this library (thanks) and noticed an open snyk CVE based on sub-sub library. 
Any thoughts on being able to make the switch from optimist to yargs?

**Comment by icedawn** [comment]
> I'm just came across this library (thanks) and noticed an open snyk CVE based on sub-sub library.
> Any thoughts on being able to make the switch from optimist to yargs?

Hey Mike, yeah, I switched a few projects over to yargs recently, so I think it's pretty easy to migrate.  I'm slammed today and tomorrow, but I can take a crack at it and make a PR unless you've already done it ;-)

**Comment by mikelax** [comment]
Cool thanks. Sorry haven't done the work yet, it's been a while since I last used yargs.

**Comment by hildjj** [comment]
I should have time in a few weeks to finish this up if it hasn't been by then.  If someone gives me a clean patch with tests, I'll merge it quickly.

**Comment by icedawn** [comment]
All good gents.  I'll see if I can get that clean patch done -- you all have plenty to do ;-)

**Comment by hildjj** [comment]
I'm doing a refresh on the library now.  Let me see if I can rebase this and get it into the release.

**Comment by hildjj** [comment]
I've got a rebase on the `apikey` branch if you want to double-check it.

**Comment by hildjj** [comment]
subsumed in #13

**Pull request closed without merging**